docs/monitors/elasticsearch-query.md

```diff
@@ -0,0 +1,282 @@
+<!--- GENERATED BY gomplate from scripts/docs/templates/monitor-page.md.tmpl --->
+
+# elasticsearch-query
+
+Monitor Type: `elasticsearch-query` ([Source](https://github.com/signalfx/signalfx-agent/tree/master/pkg/monitors/elasticsearch/query))
+
+**Accepts Endpoints**: **Yes**
+
+**Multiple Instances Allowed**: Yes
+
+## Overview
+
+This monitor metricizes aggregated responses from Elasticsearch. The monitor
+constructs SignalFx datapoints based on Elasticsearch aggregation types and
+also aggregation names.
+
+An simple configuration looks like the following:
+
+```yaml
+monitors:
+- type: elasticsearch-query
+ host: localhost
+ port: 9200
+ index: <name_of_index>
+ elasticsearchRequest: |
+    {
+      "query" : {
+        "range" : {
+          "@timestamp" : {
+            "gte": "now-5m"
+          }
+        }
+      },
+      "aggs": {
+        "avg_cpu_utilization": {
+          "avg": {
+            "field": "cpu_utilization"
+          }
+        }
+      }
+    }
+```
+
+The `elasticsearchRequest` take in a request in the format specified
+[here] (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html).
+
+The above query is performed against an index that has documents that take the following form
+
+```
+{
+   'cpu_utilization':87,
+   'memory_utilization':94,
+   'host':'helsniki',
+   'service':'android',
+   'container_id':'macbook',
+   '@timestamp':1580321240579
+}
+```
+
+This query returns the average value of `cpu_utilization` across all documents with a `@timestamp`
+in the last five minutes. This value is metricized to the following form in SignalFx -
+
+```
+{
+metric_name: avg,
```

Comment:
Shouldn't this be `avg_cpu_utilization` since that is the aggregation name?  `avg` is the metric aggregation type -- it is way too generic and meaningless to be the metric name.